completion_aggregator/fields.py

```diff
@@ -0,0 +1,67 @@
+"""
+Custom django fields.
+"""
+
+import six
+from opaque_keys.edx.keys import CourseKey, UsageKey
+from opaque_keys import InvalidKeyError
+
+from django.db import models
+
+
+class OpaqueKeyField(models.Field):
+    """
+    Abstract class for defining database fields that represent
+    opaque keys types
+    """
+    key_type = None
+
+    def __init__(self, *args, **kwargs):
+        kwargs['max_length'] = 255
+        super(OpaqueKeyField, self).__init__(*args, **kwargs)
+
+    def deconstruct(self):
+        name, path, args, kwargs = super(OpaqueKeyField, self).deconstruct()
+        del kwargs['max_length']
+        return name, path, args, kwargs
+
+    def db_type(self, connection):
+        return "VARCHAR"
```

Comment:
I got migration errors for this:
```
django.db.utils.ProgrammingError: (1064, "You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'NOT NULL, `aggregation_name` varchar(255) NOT NULL, `block_key` VARCHAR NOT NULL' at line 1")
```

But migrations ran fine if I replaced this method with:
```python
    def get_internal_type(self):
        return 'CharField'
```

ref: [Emulating Built-in Field Types](https://docs.djangoproject.com/en/1.9/howto/custom-model-fields/#emulating-built-in-field-types):

Reply:
I ran into other issues with type conversions when I tried to use CharField as the internal type.  CharField made other assumptions about the type.  I think I'll resolve it by specifying the max_length in the db_type.